pandas/core/series.py

```diff
@@ -166,9 +166,9 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     else:
                         index = Index(_try_sort(data))
                 try:
-                    if isinstance(index, DatetimeIndex):
+                    if isinstance(index, DatetimeIndex) and lib.infer_dtype(data) != 'datetime64':
```

Comment:
Before I added `.value` at line 171, `lib.fast_multiget` was throwing a `TypeError`. Now, the `else` clause must handles the conversion of datetime64 dicts because we cannot compare a `Timestamp` with a `datetime64`:

```python
>>> d = {datetime.datetime(2015, 1, 7, 2, 0): 42544017.198965244}
>>> i = pandas.core.index.Index(d)
>>> i.astype('O').values
array([Timestamp('2015-01-07 02:00:00')], dtype=object)
>>> datetime.datetime(2015, 1, 7, 2, 0) == pandas.lib.Timestamp('2015-01-07 02:00:00')
True

>>> d = {numpy.datetime64('2015-01-06T19:00:00.000000000-0500'): 42544017.198965244}
>>> i = pandas.core.index.Index(d)
>>> i.astype('O').values
array([Timestamp('2015-01-07 00:00:00')], dtype=object)
>>> numpy.datetime64('2015-01-06T19:00:00.000000000-0500') == pandas.lib.Timestamp('2015-01-07 00:00:00')
False
```